### ISSUE TITLE
[bug]: Modal does not show items in 1.0.0-rc.13

### ISSUE THREAD

**Issue by nhatnguyenbp** [WalletConnect/modal-react-native#96]
### Description

After updating wallet connect modal from 1.0.0-rc.11 to 1.0.0-rc.13, I can not see any items in the modal, It always show a loading indicator. I check the logs and it throw error: 

`TypeError: ownKeys target is non-extensible but key is missing from trap result`

### WalletConnect Modal SDK version

1.0.0-rc.13

### Output of `npx react-native info`

System:
  OS: macOS 14.1
  CPU: (8) arm64 Apple M2
  Memory: 65.72 MB / 16.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 18.12.1
    path: /usr/local/bin/node
  Yarn:
    version: 1.22.19
    path: /usr/local/bin/yarn
  npm:
    version: 9.1.2
    path: /usr/local/bin/npm
  Watchman:
    version: 2023.02.27.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.13.0
    path: /usr/local/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.0
      - iOS 17.0
      - macOS 14.0
      - tvOS 17.0
      - watchOS 10.0
  Android SDK:
    API Levels:
      - "29"
      - "30"
      - "31"
      - "32"
      - "33"
    Build Tools:
      - 30.0.2
      - 30.0.3
      - 31.0.0
      - 33.0.0
      - 33.0.1
      - 34.0.0
    System Images:
      - android-22 | Google APIs ARM 64 v8a
      - android-29 | Google APIs ARM 64 v8a
      - android-33 | Google APIs ARM 64 v8a
    Android NDK: Not Found
IDEs:
  Android Studio: 2022.3 AI-223.8836.35.2231.10406996
  Xcode:
    version: 15.0.1/15A507
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 18.0.2.1
    path: /usr/bin/javac
  Ruby:
    version: 2.6.10
    path: /usr/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.72.5
    wanted: 0.72.5
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: false

### Steps to reproduce

- Update wallet connect to .rc.13
- Press on connect button
- Observe the modal showing

### Snack, code example, screenshot, or link to a repository

![image](https://github.com/WalletConnect/modal-react-native/assets/80235511/25c14b4a-9f5b-4d54-ae91-a8deee4fcc8b)

**Comment by ignaciosantise** [comment]
@nhatnguyenbp thanks for reporting. I couldn't reproduce the issue but found that same error in `valtio` repository https://github.com/pmndrs/valtio/issues/765

Can you check if the new latest version of valtio solves the issue?
You can check this by adding this in your package.json file:

```
"resolutions": {
    "valtio": "1.12.0"
  },
```

If it doesn't work, i'll probably downgrade the version again until they fix it

**Comment by ignaciosantise** [comment]
@nhatnguyenbp also, could you provide a public repository with the reproducible issue?

**Comment by nhatnguyenbp** [comment]
Thanks for your response @ignaciosantise. Let me check on my side, I will split code to public project if needed.

**Comment by nhatnguyenbp** [comment]
@ignaciosantise  Downgrade valtio to 1.10.5 by put it to `overrides` worked for me. I also tried with update `react-native 0.72.6`, update `"valtio": "1.12.0"`, put the valtio to `resolutions` but they did not work. 
```
"overrides": {
    "valtio": "1.10.5"
  },
  ````

**Comment by ignaciosantise** [comment]
@nhatnguyenbp thanks for the response. I've downgraded valtio's version until they fix the issue. 

This should be fixed in `1.0.0-rc.14`
